Add ignoreCommitsOlderThan date limiting

grandchild opened the pull request:
I would like to be able to see only commits with a limited age (or commits after some date).

This pull request implements a new setting that goes along the line of _ignoreCommitsIfMessageMatches_:
 _ignoreCommitsIfOlderThan_.

`settings:`
```yaml
{
    # ...
    "dateFormat": "YYYY-MM-dd HH:mm:ss",
    "ignoreCommitsIfOlderThan": "2017-01-01 00:00:00",
    # ...
}
```

The rationale is that perhaps projects might:
* use a git-based changelog to inform coworkers, rather than customers, making a more short-lived "news"-style log desirable, _(this is our use-case)_
* not (yet?) use frequent-enough tags in their repository, _(also true where I work)_
* want to provide a sense for the liveliness of a project

There are tests.

What do you guys think? Any suggestions? Anything I have overlooked, or that should be improved? Would be very happy to see this merged. :smile:

One point of critique could be that I coupled the date format of `ignoreCommitsIfOlderThan` to the quite unrelated `dateFormat`, but it does eliminate the need for an extra setting. Alternatively one could think about simply mandating a specific format like `YYYY-MM-dd HH:mm:ss` or so, ignoring `dateFormat`.

tomasbjerre commented:
I will try to find time to look at this as soon as possible! Definitely like the feature!

grandchild commented:
Okay, thanks for letting me know, and sure, take your time!
And don't be shy with criticism, if you want stuff added, changed, removed, renamed, etc. -- just say it :)

tomasbjerre commented:
I made a couple of changes, see my commit in master: https://github.com/tomasbjerre/git-changelog-lib/commits/master

What do you think? I'll add it to the plugins if you are ok with it.

grandchild commented:
Looks good! Thanks for the cleanup, using Date is much nicer, of course. In my mind was thinking about the gradle-plugin usage, so I didn't consider it.

And yes, definitely bind it to the plugin. I'm particularly interested in the gradle plugin as you can see. Also, I'm a fan of CC0, so consider everything as if it were your own code, do whatever you feel like :)

tomasbjerre commented:
This is now released in the gradle plugin!

grandchild commented:
Super, thanks!
